File: typescript/src/app/04-i-interface-segregation/concept/after/HpAllInOne.ts
import ICopier from './ICopier';
import IFax from './IFax';
import IPrinter from './IPrinter';
import IScanner from './IScanner';

export default class HpAllInOne implements IPrinter, IScanner, ICopier, IFax {
  print(): string {
    return 'Printing done!';
  }
  scan(): string {
    return 'Scanning done!';
  }
  copy(): string {
    return 'Copying done!';
  }
  fax(): string {
    return 'Faxing done!';
  }
}

File: java/src/main/java/isp/before/HpAllInOne.java
package isp.before;

public class HpAllInOne implements AllInOne {
    @Override
    public String print() {
        return "Print done!";
    }

    @Override
    public String scan() {
        return "Scan done!";
    }

    @Override
    public String copy() {
        return "Copy done!";
    }

    @Override
    public String fax() {
        return "Fax done!";
    }
}
